plugins/kn-curl/kn-curl

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+# Copyright © 2020 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+USAGE=$"Usage:
+  kn curl NAME
+
+Examples:
+  # Curl service hello
+  kn curl hello
+
+  # Curl service hello in namespace myns
+  kn curl hello -n myns
+
+  Flags:
+	-n, --namespace string             Specify the namespace to operate in.
+"
+
+# Need at least the service name
+if [ -z "$1" ]; then
+	echo "$USAGE"
+	exit 0
+fi
+
+# If -h or --help is first argument
+if [ "$1" == "-h" ] || [ "$1" == "--help" ]; then
+  echo "$USAGE"
+  exit 0
+fi
+
+# Flags
+N=$"-n"
+NAMESPACE=$"--namespace"
+NS="default"
+
+# Parse arguments for -n or --namespace and namespace value
+if [ "$N" == "$2" ] | [ "$NAMESPACE" == "$2" ]; then
+  NS="$3"
+fi
+
+# Get kubectl path
+KUBECTL=`which kubectl`
+if [ -z "$KUBECTL" ]; then
+  echo "$USAGE"
+  echo "kubectl must also be in your PATH"
+  exit 0
+fi
+
+# Get INGRESS assuming Istio
+INGRESS=$($KUBECTL get svc istio-ingressgateway --namespace istio-system | awk '{print $4}' | tail +2)
```

Comment:
what's about Knative installations that don't use Istio but e.g. Kourier, Gloo or Contour as ingress ? (no unusual these days)

Reply:
At least I would verify the result to check whether an istio gateway is really available. And if not, don't go into the fallback below but stop with an error.

Reply:
I am thinking about how to do it for other ingress. See my TODOs. I was thinking of just trying each in order. But I need to get a test system with Contour or Gloo setup first.

Reply:
Done